src/index.ts

```diff
@@ -34,6 +35,30 @@ async function startServer(): Promise<void> {
     case 'http': {
       const { url } = await startExpressServer({ basePath: serverName, config, logLevel });
 
+      if (!config.oauth.enabled) {
+        console.warn(
+          '⚠️ TRANSPORT is "http" but OAuth is disabled! Your MCP server may not be protected from unauthorized access!',
+        );
+
+        const rl = readline.createInterface({
+          input: process.stdin,
+          output: process.stdout,
+        });
+
+        const input = (
+          await rl.question('⚠️ To accept these risks, please type "I accept" and press Enter.\n> ')
+        )
+          .toLocaleLowerCase()
+          .replaceAll('"', '')
+          .trim();
```

Comment:
I might need to change this approach. Admins may not have terminal access to the server they are deploying to and can't type anything. Instead, I might rename the `DISABLE_OAUTH` environment variable to `DANGEROUSLY_DISABLE_OAUTH`.